### PR TITLE
[SQL-DS-CACHE-126][POAE7-1132] fix buffer overwritten when prefetching batches from APE server

### PR DESCRIPTION
**Pull request by iyupeng**
## What changes were proposed in this pull request?

* fix answer errors caused by buffer overwritten in APE server

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

integration tests

### CODE REVIEW

File: oap-ape/ape-java/ape-server/src/main/java/com/intel/oap/ape/service/netty/server/RequestHandler.java [hunk 1]
```diff
@@ -184,24 +195,30 @@ private void handleParquetReaderInitRequest(NettyMessage.ParquetReaderInitReques
...
+            // check next batch
+            hasNextBatch = ParquetReaderJNI.hasNext(reader);
+        }
 
         // header info
         boolean[] compositeFlags = new boolean[columnCount];
         int[] dataBufferLengths = new int[columnCount];
         int compositedElementCount = 0;
-        ByteBuf compositedElementLengths = ctx.alloc().ioBuffer(4 * rowCount);
+        ByteBuf compositedElementLengths = channel.alloc().ioBuffer(4 * rowCount);
```
Comment (jikunshang) [inline oap-ape/ape-java/ape-server/src/main/java/com/intel/oap/ape/service/netty/server/RequestHandler.java]: why 4 here? try to avoid magic number?

**Review by jikunshang — APPROVED** [review]
Overall LGTM. cc @yma11

File: oap-ape/ape-java/ape-server/src/main/java/com/intel/oap/ape/service/netty/server/RequestHandler.java (hunk 1 above)
Comment (iyupeng) [reply oap-ape/ape-java/ape-server/src/main/java/com/intel/oap/ape/service/netty/server/RequestHandler.java]: It's to set the initial capacity of a buffer. This buffer is used to store lengths of byte arrays. So all elements will be integers.
`4 * rowCount` means that the initial capacity is a length that can hold 1 column of elements.

Comment (jikunshang) [reply oap-ape/ape-java/ape-server/src/main/java/com/intel/oap/ape/service/netty/server/RequestHandler.java]: Got it. Thanks.

**Pull request merged**